app/src/src_nonfreenet/org/breezyweather/sources/bmkg/BmkgService.kt

```diff
@@ -0,0 +1,286 @@
+        val ibf = mutableListOf<Observable<BmkgIbfResult>>()
+        for (day in 1..3) {
+            ibf.add(
+                if (requestedFeatures.contains(SecondaryWeatherSourceFeature.FEATURE_ALERT)) {
+                    mApi.getIbf(
+                        lat = location.latitude,
+                        lon = location.longitude,
+                        day = day,
+                        api_token = apiKey
+                    ).onErrorResumeNext {
+                        Observable.create { emitter ->
+                            emitter.onNext(BmkgIbfResult())
+                        }
+                    }
+                } else {
+                    Observable.create { emitter ->
+                        emitter.onNext(BmkgIbfResult())
+                    }
+                }
+            )
+        }
+
+        return Observable.zip(current, warning, ibf[0], ibf[1], ibf[2]) {
+                currentResult: BmkgCurrentResult,
+                warningResult: BmkgWarningResult,
+                ibf1Result: BmkgIbfResult,
+                ibf2Result: BmkgIbfResult,
+                ibf3Result: BmkgIbfResult
+            ->
+            convertSecondary(
+                context = context,
+                currentResult = if (requestedFeatures.contains(SecondaryWeatherSourceFeature.FEATURE_CURRENT)) {
+                    currentResult
+                } else null,
+                warningResult = if (requestedFeatures.contains(SecondaryWeatherSourceFeature.FEATURE_ALERT)) {
+                    warningResult
+                } else null,
+                ibf1Result = if (requestedFeatures.contains(SecondaryWeatherSourceFeature.FEATURE_ALERT)) {
+                    ibf1Result
+                } else null,
+                ibf2Result = if (requestedFeatures.contains(SecondaryWeatherSourceFeature.FEATURE_ALERT)) {
+                    ibf2Result
+                } else null,
+                ibf3Result = if (requestedFeatures.contains(SecondaryWeatherSourceFeature.FEATURE_ALERT)) {
+                    ibf3Result
+                } else null
+            )
+        }
+    }
+
+    override fun requestReverseGeocodingLocation(
+        context: Context,
+        location: Location
+    ): Observable<List<Location>> {
+        val locationList = mutableListOf<Location>()
+        return mApi.getLocation(
+            lat = location.latitude,
+            lon = location.longitude
+        ).map {
+            locationList.add(convert(location, it))
+            locationList
+        }
+    }
+
+    companion object {
+        private const val BMKG_BASE_URL = "https://cuaca.bmkg.go.id/"
+        // The API key can be found from tokenDefault in the HTML source of https://cuaca.bmkg.go.id/
+        private const val BMKG_API_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjFjNWFkZWUxYzY5MzM0NjY2N2EzZWM0MWRlMjBmZWZhNDcxOTNjYzcyZDgwMGRiN2ZmZmFlMWVhYjcxZGYyYjQiLCJpYXQiOjE3MDE1ODMzNzl9.D1VNpMoTUVFOUuQW0y2vSjttZwj0sKBX33KyrkaRMcQ"
```

Comment:
Needs to be implemented as a build config `BMKG_KEY`. `.github/workflows/push.yml` needs an update to add the key there.

Reply:
Fixed.